### PR TITLE
CORE-8672 Storing relevancy information

### PR DESCRIPTION
**Pull request by mbrkic-r3**
- Added `relevantStates` to `PersistTransaction` ledger Avro schema
- Removed column `utxo_transaction_relevancy.is_relevant`

### CODE REVIEW

**Review by lankydan — DISMISSED** [review]

File: data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/ComponentPosition.avsc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "type": "record",
+  "name": "ComponentPosition",
+  "doc": "The group index and the leaf index of the transaction component",
+  "namespace": "net.corda.data.ledger.persistence",
+  "fields": [
+    {
+      "name": "groupIndex",
```
Comment (blsemo) [inline data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/ComponentPosition.avsc]: Do we need the group index here? Shouldn't that always be OutputStates?

File: data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml [hunk 2]
```diff
@@ -166,9 +166,6 @@
             <column name="leaf_idx" type="INT">
                 <constraints nullable="false"/>
             </column>
-            <column name="is_relevant" type="BOOLEAN">
-                <constraints nullable="false"/>
-            </column>
```
Comment (blsemo) [inline data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:171]: So we don't store it this way, but where is the schema for storing the relevancy information in the new format? What did I miss?

Comment (blsemo) [reply data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:171]: Reading the code - do we reuse the tx output table to only record relevant states in it?

Comment (mbrkic-r3) [reply data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:171]: My understanding was that we only insert into table `utxo_transaction_relevancy` when component is relevant, in which case this field would always be `true`, isn't that the case?

File: data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/ComponentPosition.avsc (hunk 1 above)
Comment (mbrkic-r3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/ComponentPosition.avsc]: I was not aware that is always fixed, I'll change it

File: data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml (hunk 2 above)
Comment (blsemo) [reply data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:171]: Ok, that makes sense.

Comment (mbrkic-r3) [reply data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:171]: > Reading the code - do we reuse the tx output table to only record relevant states in it?

Tx output table is not used at the moment.

File: data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/ComponentPosition.avsc (hunk 1 above)
Comment (mbrkic-r3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/ComponentPosition.avsc]: fixed

**Review by blsemo — DISMISSED** [review]
LGTM

**Review by blsemo — APPROVED** [review]
Re-approved

**Pull request merged**